0xPolygon/polygon-edge
# Update Staking SC bytecode for PR#14 in staking-contract

# Description

Please provide a detailed description of what was done in this PR

This PR updates Staking SC bytecode based on latest implementation in https://github.com/0xPolygon/staking-contracts/pull/14

# Changes include


# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist


## Testing


### Manual tests

1. Please setup PoS chain and specify `min-validator-count`, `max-validator-count` to the same number of initial validators and premine balance for an account.

```bash
go run ./main.go genesis --consensus ibft --ibft-validators-prefix-path test-chain- --pos --min-validator-count 4 --max-validator-count 4 \
--premine 0x... \
--bootnode /ip4/127.0.0.1/tcp/11001/p2p/... \
--bootnode /ip4/127.0.0.1/tcp/12001/p2p/... \
--bootnode /ip4/127.0.0.1/tcp/13001/p2p/... \
--bootnode /ip4/127.0.0.1/tcp/14001/p2p/... 
```

2. Start PoS chain
3. Stake less amount than the threshold for validator from the premined account. Please check this tx is successful.
4. Stake more amount than the threshold for validator from the premined account. Please check this tx fails.
5. Unstake from the premined account. Please check this tx is successful.

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it
